### ISSUE TITLE
Bug - Link automatically disconnecting and reconnecting on site.

### ISSUE THREAD

**Issue by Brennok** [civitai/civitai-link-desktop#25]
- Win 11 Pro with both Firefox and Chrome

- I didn't have the issue with constant flapping issue on the civitai extension, thought it does happen occasionally, but I did have the issue where one day it just stopped showing what was installed and only once after that, the day my membership renewed, it showed again. I want to say I haven't seen the models marked installed since back towards the end of February.  

- No issues on the app site. 

I also tried all addons disabled with no luck so nothing tripping it that I can find.

Of course now that I say it I was just disconnected and not reconnected with the extension.

_Originally posted by @Brennok in https://github.com/civitai/civitai-link-desktop/issues/21#issuecomment-2048631914_

I see these errors in Firefox
Firefox can’t establish a connection to the server at wss://link.civitai.com/api/socketio/?EIO=4&transport=websocket&sid=CiMIr90pF1Nncge3ABk9.
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist.
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. [background.js:1](moz-extension://914a1cf4-ffb4-4e4b-90a9-601a0a5fc73b/background.js)
    sendMessageWithResponse moz-extension://914a1cf4-ffb4-4e4b-90a9-601a0a5fc73b/background.js:1

With the Civitai extension, I would usually see it join a room, is it possible the room isn't being created so it connects and drops? Also I don't see a way to change the Link code int he app so I am hesitant to delete it and generate a new one. 

I am wondering if it is possibly due to a delayed completion of the app. Usually with the extension, you plug in the Link code and it immediately connects. Due to the delayed indexing it didn't try to connect for a good few hours. Link at the time even would still let me copy the key. It is a complete guess and just throwing stuff against the wall, but didn't know if the delay here might cause issues on the server side. 

Also not sure if it is supposed to look like this but I see all the connection attempts tripled. 
roomPresence 
Object { client: 1, sd: 1 }
[civitai-link.6f46ccd0869af80d.js:1:4764](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
roomPresence 
Object { client: 1, sd: 1 }
[civitai-link.6f46ccd0869af80d.js:1:4764](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
roomPresence 
Object { client: 1, sd: 1 }
[civitai-link.6f46ccd0869af80d.js:1:4764](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
emitMessage 
Object { msg: "Stable Diffusion service connected" }
[civitai-link.6f46ccd0869af80d.js:1:2844](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
emitMessage 
Object { msg: "Stable Diffusion service connected" }
[civitai-link.6f46ccd0869af80d.js:1:2844](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
emitMessage 
Object { msg: "Stable Diffusion service connected" }
[civitai-link.6f46ccd0869af80d.js:1:2844](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
Initialized instance: 9856 3 [civitai-link.6f46ccd0869af80d.js:1:7215](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
cl-id: 9856 3 [civitai-link.6f46ccd0869af80d.js:1:3155](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
updateSharedValue 
Object { type: "instance", value: {…} }
[civitai-link.6f46ccd0869af80d.js:1:2169](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
updateSharedValue 
Object { type: "instance", value: {…} }
[civitai-link.6f46ccd0869af80d.js:1:2169](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
updateSharedValue 
Object { type: "instance", value: {…} }
[civitai-link.6f46ccd0869af80d.js:1:2169](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
roomPresence 
Object { client: 1, sd: 0 }
[civitai-link.6f46ccd0869af80d.js:1:4764](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
roomPresence 
Object { client: 1, sd: 0 }
[civitai-link.6f46ccd0869af80d.js:1:4764](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
roomPresence 
Object { client: 1, sd: 0 }
[civitai-link.6f46ccd0869af80d.js:1:4764](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
emitMessage 
Object { msg: "Stable Diffusion service disconnected" }
[civitai-link.6f46ccd0869af80d.js:1:2844](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
emitMessage 
Object { msg: "Stable Diffusion service disconnected" }
[civitai-link.6f46ccd0869af80d.js:1:2844](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)
emitMessage 
Object { msg: "Stable Diffusion service disconnected" }
[civitai-link.6f46ccd0869af80d.js:1:2844](https://civitai.com/_next/static/chunks/civitai-link.6f46ccd0869af80d.js)

**Comment by Brennok** [comment]
Just as an update with 1.14.1, I see Link loop between Disconnected, No Key, and Connected. I am not seeing any change on the website which just shows disconnected.

**Comment by cmoody** [comment]
@Brennok hopefully got some good news for you. After some investigation I think theres a limit on the size of a message that the socket can send. With large number of files it wouldnt take but 1000-2000 files to possibly hit the default limit. Ive got two things going out to try and fix this. The first is an update to the server to increase message limit to 5MB. We can tweak it pretty quick if we are seeing that messages are still too large.

The next bit is I noticed Im over sending data thats actually needed by the site. Im working on a bunch of UI updates and will hopefully get this fix out tomorrow.

Let me know if its still flapping and if not then hopefully the app fix going out will just be a little extra boost.

**Comment by Brennok** [comment]
@cmoody It worked a bit this morning without flapping, but then started up again. I sent the logs to support during the office hours twitch stream, but can attach here if you want. I definitely have more than 2k files. I think in Loras along I have 5k, Even when it wasn't flapping I still couldn't see what was installed on the site though. I can send to SD, but didn't test that to see if it worked. 

Right now after restarting Civitai Link, both installed and no flapping going on so fingers crossed.

**Comment by Brennok** [comment]
Possibly related, also emailed to support in response to an email from Ally and Max, Link closes after about an hour. Also I found it creating and possibly replacing my model preview images with empty .png preview images. Previews were downloaded prior using Civitai helper extension or Civitai extension. Tested again with Link on and no instance of Stable Diffusion running. Created 300 empty .png previews before I closed Link.
